src/gui/sdlmain.cpp

```diff
@@ -1254,39 +1290,86 @@ SDL_Rect GFX_GetSDLSurfaceSubwindowDims(uint16_t width, uint16_t height)
 	return rect;
 }
 
+// kvas kompot pelmeni shashlyki kotlety bliny belyashi
+static SDL_Rect get_canvas_size(const SCREEN_TYPES screen_type)
+{
+	int width  = 0;
+	int height = 0;
+
+	switch (screen_type) {
+	case SCREEN_SURFACE:
+		SDL_GetWindowSize(sdl.window, &width, &height);
+		break;
+	case SCREEN_TEXTURE:
+		if (SDL_GetRendererOutputSize(sdl.renderer, &width, &height) != 0)
+			LOG_ERR("SDL: Failed to retrieve output size: %s",
+			        SDL_GetError());
+		break;
+#if C_OPENGL
+	case SCREEN_OPENGL:
+		SDL_GL_GetDrawableSize(sdl.window, &width, &height);
+		break;
+#endif
+	}
+
+	return SDL_Rect{0, 0, width, height};
+}
+
 static SDL_Window *setup_window_pp(SCREEN_TYPES screen_type, bool resizable)
 {
 	assert(sdl.window);
 
-	int w, h;
-	if (sdl.desktop.fullscreen) {
-		SDL_GetWindowSize(sdl.window, &w, &h);
-	} else {
-		w = sdl.desktop.requested_window_bounds.width;
-		h = sdl.desktop.requested_window_bounds.height;
+	int window_width  = 0;
+	int window_height = 0;
+	SDL_GetWindowSize(sdl.window, &window_width, &window_height);
+	assert(window_width > 0 && window_height > 0);
+
+	const auto canvas = get_canvas_size(screen_type);
+	assert(canvas.w > 0 && canvas.h > 0);
```

Comment:
Suggest moving the assert into `get_canvas_size`. 
When we're in control of our own functions, we can put the guarantees in one place.

Reply:
Done.